### PR TITLE
grafana: update node_exporter deployment for free space monitoring

### PR DESCRIPTION
**Pull request by consideRatio**
Does not close this issue fully, but to a large extent.
- #2098 

## Demonstrated to work on ubc-eoas staging

image

## Followup actions by me
- Run grafana workflow
- Communicate in related support ticket

### CODE REVIEW

File: helm-charts/basehub/templates/home-space-reporter.yaml [hunk 1]
```diff
@@ -1,57 +1,67 @@
...
+            - --web.listen-address=:9100
+          ports:
+            - containerPort: 9100
+              name: metrics
+              protocol: TCP
+          securityContext:
+            allowPrivilegeEscalation: false
+            runAsGroup: 65534
+            runAsNonRoot: true
+            runAsUser: 65534
```
Comment (GeorgianaElena) [inline helm-charts/basehub/templates/home-space-reporter.yaml:50]: @consideRatio can you please share the motivation behind moving these security context settings from a deployment level, to a container one? I understand what this does, but I'm curious to understand if this part of a "best practices" or has a some other motivation?

**Review by GeorgianaElena — APPROVED** [review]
This looks good to me @consideRatio! Thank you

File: helm-charts/basehub/templates/home-space-reporter.yaml (hunk 1 above)
Comment (consideRatio) [reply helm-charts/basehub/templates/home-space-reporter.yaml:50]: Unless we have side-car or init containers added, this is equivalent. If we do, we didn't add them, so then its better if they aren't influenced by a podSecurityPolicy I think.

I've seen a pod's securityPolicy break assumptions for istio side-car containers added automatically in the past.

I made a comment in the commit adding this to Yuvi's PR: https://github.com/jupyterhub/grafana-dashboards/pull/51/commits/9db9ca8c4a15d0a20697eecad4e5d020c552e6e0

---

Note that this is a copy-paste mostly from the grafana-dashboards, so we are more in sync with whats documented there.

File: helm-charts/basehub/templates/home-space-reporter.yaml [hunk 2]
```diff
@@ -1,57 +1,67 @@
...
-# we have for our hub mounted so we can monitor free space usage.
 {{- if or .Values.nfs.enabled .Values.azureFile.enabled }}
+# To provide data for the jupyterhub/grafana-dashboards dashboard about free
+# space in the shared volume, which contains users home folders and the shared
+# folder, we deploy prometheus node-exporter to collect this data for prometheus
+# server to scrape.
+#
+# This is based on the Deployment manifest in jupyterhub/grafana-dashboards'
+# readme: https://github.com/jupyterhub/grafana-dashboards#additional-collectors
+#
```
Comment (consideRatio) [inline helm-charts/basehub/templates/home-space-reporter.yaml:9]:
```suggestion
# To provide data for the jupyterhub/grafana-dashboards dashboard about free
# space in the shared volume, which contains users home folders etc, we deploy
# prometheus node-exporter to collect this data for prometheus server to scrape.
#
# This is based on the Deployment manifest in jupyterhub/grafana-dashboards'
# readme: https://github.com/jupyterhub/grafana-dashboards#additional-collectors
#
```

File: helm-charts/basehub/templates/home-space-reporter.yaml (hunk 1 above)
Comment (GeorgianaElena) [reply helm-charts/basehub/templates/home-space-reporter.yaml:50]: Makes more sense now. Thanks @consideRatio

File: helm-charts/basehub/templates/home-space-reporter.yaml [hunk 3]
```diff
@@ -1,57 +1,67 @@
...
   template:
     metadata:
       annotations:
+        # This enables prometheus to actually scrape metrics from here
         prometheus.io/scrape: "true"
         prometheus.io/port: "9100"
       labels:
         app: jupyterhub
-        component: home-metrics
+        # This component label is used in our dashboard, so do not remove
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply helm-charts/basehub/templates/home-space-reporter.yaml:31]:
```suggestion
        # The component label below should match a grafana dashboard definition
        # in jupyterhub/grafana-dashboards, do not change it!
```

The top of the file already specifies what dashboard definition it is, and outlining that again is too much detail for this comment I think.

**Review by consideRatio — COMMENTED** [review]
I opened https://github.com/jupyterhub/grafana-dashboards/pull/55 to match the changes I suggest above, so that we are in parity with the jupyterhub/grafana-dashboards provided example and also upstream your clarifications there @GeorgianaElena!

Thank you for the review, :+1: on your notes!

**Pull request merged**